### ISSUE TITLE
Can't reassign has_and_belongs_to_many associations of superseded records

### ISSUE THREAD

**Issue by joeroe** [xronos-ch/xronos.rails#188]
Because ActiveRecord::Reflection doesn't appear to give us any explicit information on the join table used.

Quick solution: write a method to infer the join table's name and update it. The problem is this leaves us with no way to record the change (we can't use paper_trail with the join table).

Probably better solution: don't use has_and_belongs_to_many (https://petehanner.medium.com/the-siren-song-of-has-and-belongs-to-many-in-rails-f526d6a6be79). If there is an explicitly class in the middle, the problem goes away.

Currently this would only relevant for the association between sites and site types, so perhaps reconsider this?

**Comment by MartinHinz** [comment]
I agree, and I like solution 2 better. Would SiteClassification be a model that would work here?

**Comment by joeroe** [comment]
Solved by #515
